### ISSUE TITLE
Dynamically Update Assets (height field)

### ISSUE THREAD

**Issue by vikashplus** [deepmind/mujoco#958]
I'm trying to dynamically update assets. While the model is getting updated and the simulation reflects the update, the viewer is not appropriately reflecting the update despite these updates being pushed to the mjr_contexts. Likely I'm missing something. Im attaching a simple script to reproduce.

```python 
from mujoco import viewer
from dm_control.mujoco import Physics
from dm_control.mujoco.wrapper.mjbindings import mjlib
import numpy as np

# Prepare physics
sim = Physics.from_xml_path('height_field.xml')

# Prepare viewer
window = viewer.launch_passive(sim.model.ptr, sim.data.ptr)

# New height field
for _ in range(5):

    # Prep new hfield
    sim.model.hfield_data[:] = np.random.uniform(high=.2, low=.1, size=np.shape(sim.model.hfield_data))

    # Re-Upload hfield data to gpu
    if sim.contexts:
        print("Uploading new asset")
        with sim.contexts.gl.make_current() as ctx:
            ctx.call(mjlib.mjr_uploadHField, sim.model.ptr, sim.contexts.mujoco.ptr, 0)

    # Sync viewer
    window.sync()

    # wait for user
    input("Press enter to generate new height field")
```

mujoco==2.3.5
dm-control==1.0.11
os = Ubuntu 18.04.6 LTS
Python 3.8.16

[asset_update.zip](https://github.com/deepmind/mujoco/files/11899330/asset_updates.zip)

cc: @cameronberg

**Comment by saran-t** [comment]
This cannot work since the viewer is completely separate from the Python process.

**Comment by saran-t** [comment]
I'll keep thinking about alternative solutions.

**Comment by vikashplus** [comment]
This is amazing! Thanks @saran-t 
I'll give it a try soon.

**Comment by saran-t** [comment]
This is now available in version 2.3.7 which was just released.
